VAMobile/src/utils/waygateConfig.ts

```diff
@@ -0,0 +1,170 @@
+  enabled: true,
+  errorMsgTitle: undefined,
+  errorMsgBody: undefined,
+  appUpdateButton: false,
+  allowFunction: false,
+  denyAccess: false,
+}
+
+export let waygateConfig: WaygateToggleValues = {
+  WG_HomeScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_ProfileScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_PersonalInformationScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_HowDoIUpdateScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_PreferredNameScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_GenderIdentityScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_WhatToKnowScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_ContactInformationScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_HowWillYouScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_EditAddressScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_EditPhoneNumberScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_EditEmailScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_MilitaryInformationScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_IncorrectServiceInfoScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_SettingsScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_ManageYourAccountScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_NotificationsSettingsScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_ContactVAScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_VeteransCrisisLineScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_VeteranStatusScreen: JSON.parse(JSON.stringify(waygateDefault)),
+  WG_LoginScreen: JSON.parse(JSON.stringify(waygateDefault)),
+}
+
+export const waygateEnabled = (feature: WaygateToggleType): Waygate => {
+  if (overrideRemote) {
+    return waygateConfig[feature]
+  } else {
+    const waygate = remoteConfig().getValue(feature)?.asString()
+    if (waygate) {
+      return JSON.parse(waygate) as Waygate
+    } else {
+      return {
+        enabled: true,
+        errorMsgTitle: undefined,
+        errorMsgBody: undefined,
+        appUpdateButton: false,
+        allowFunction: false,
+        denyAccess: false,
+      }
+    }
+  }
+}
+
+export const loadWaygateOverrides = async (): Promise<void> => {
+  try {
+    const waygateOverrides = await AsyncStorage.getItem(WAYGATE_OVERRIDES_KEY)
+    if (waygateOverrides) {
+      setOverrideRemote(true)
+      waygateConfig = JSON.parse(waygateOverrides) as WaygateToggleValues
+    }
+  } catch (err) {
+    logNonFatalErrorToFirebase(err, 'loadOverrides: AsyncStorage error')
+  }
+}
+
+export const waygateNativeAlert = (feature: WaygateToggleType): boolean => {
+  const waygate = waygateEnabled(feature)
+  if (waygate.enabled === false && waygate.denyAccess === true && waygate.errorMsgTitle) {
+    Alert.alert(waygate.errorMsgTitle, waygate.errorMsgBody, [
+      {
+        text: 'OK',
+        style: 'cancel',
+      },
+    ])
+    return false
```

Comment:
I've added some documentation to the function so a hover over will clarify the false/true paradigm of the function for now. Additionally since this isn't directly related to a waygate configuration in firebase we can make edits as we please.